### PR TITLE
Log import progress and update job status.

### PR DESCRIPTION
**Pull request by labkey-jeckels**
Creating a PR as an easier way to comment

### CODE REVIEW

File: src/org/labkey/targetedms/SkylineDocImporter.java [hunk 1]
```diff
@@ -2418,4 +2486,182 @@ public OptimizationInfo(boolean insertCEOptmizations, boolean insertDPOptmizatio
...
+                _partPercent = partPercent;
+                _progressMonitor = progressMonitor;
+            }
+            public int getPartPercent()
+            {
+                return _partPercent;
+            }
+            public void setPartPercent(int partPercent)
+            {
+                _partPercent = partPercent > 100 ? 100 : partPercent;
```
Comment (labkey-jeckels) [inline src/org/labkey/targetedms/SkylineDocImporter.java]: Math.min() is a little easier to read

File: src/org/labkey/targetedms/SkylineDocImporter.java [hunk 2]
```diff
@@ -2418,4 +2486,182 @@ public OptimizationInfo(boolean insertCEOptmizations, boolean insertDPOptmizatio
...
+        private final ProgressPart _parserPart;     // Document parsing
+        private final ProgressPart _qcCleanupPart;  // QC cleanup - remove redundant samples from older runs
+        private final ProgressPart _foldChangePart; // Fold change calculations for group comparisons
+        private final ProgressPart _calCurvesPart;  // Calibration curve calculations
+
+        private List<ProgressPart> _progressParts;
+        private final PipelineJob _job;
+        private int _lastProgressPerc = 0;
+
+        private final Logger LOG;
```
Comment (labkey-jeckels) [inline src/org/labkey/targetedms/SkylineDocImporter.java]: This isn't static, so I'd name it _log instead

File: src/org/labkey/targetedms/SkylineDocImporter.java [hunk 3]
```diff
@@ -410,7 +433,6 @@ private ReplicateInfo insertReplicates(TargetedMSRun run, SkylineDocumentParser
 
             if (folderType == TargetedMSService.FolderType.QC)
             {
-                // In QC folders insert a replicate only if at least one of the associated sample files will be inserted
```
Comment (labkey-jeckels) [inline src/org/labkey/targetedms/SkylineDocImporter.java]: I see what you mean about the current behavior - I retract my previous defense of this comment. :)

File: src/org/labkey/targetedms/SkylineDocImporter.java [hunk 4]
```diff
@@ -2308,28 +2358,32 @@ int quantifyRun(TargetedMSRun run, PeptideSettings pepSettings, Collection<Group
...
             return 0;
         }
 
+        _log.info("Quantifying run...");
         RunQuantifier quantifier = new RunQuantifier(run, _user, _container);
+        int i = 0;
+        IProgressStatus _foldChangeStatus = _progressMonitor.getFoldChangeProgressTracker();
         for (GroupComparisonSettings groupComparison : groupComparisons)
         {
+            _log.info(String.format("Calculating fold change for group comparison %s, %d of %d", groupComparison.getName(), ++i, groupComparisons.size()));
```
Comment (labkey-jeckels) [inline src/org/labkey/targetedms/SkylineDocImporter.java]: Consider kicking this down to debug-level

File: src/org/labkey/targetedms/SkylineDocImporter.java [hunk 5]
```diff
@@ -461,6 +470,35 @@ private ReplicateInfo insertReplicates(TargetedMSRun run, SkylineDocumentParser
...
+    {
+        int total = replicateInfo.oldSamplesToDelete.values().stream().mapToInt(List::size).sum();
+        int s = 0;
+        IProgressStatus status = _progressMonitor.getQcCleanupProgressTracker();
+        for(Map.Entry<String, List<SampleFile>> entry: replicateInfo.oldSamplesToDelete.entrySet())
+        {
+            for (SampleFile existingSample : entry.getValue())
+            {
+                SampleFile srcFile = TargetedMSManager.deleteSampleFileAndDependencies(existingSample.getId());
+                _log.info("Updating previously imported data for sample file " + entry.getKey() + " in QC folder.");
```
Comment (labkey-jeckels) [inline src/org/labkey/targetedms/SkylineDocImporter.java]: This could potentially be debug level

**Comment by vagisha** [comment]
Thanks for the comments.  I made the recommended changes in the previous commit, and made a few more logging changes.  Let me know if there is anything else.

**Comment by labkey-jeckels** [comment]
@vagisha looks good to me. I opened the PR to add my comments, so I can't approve it. But you should be able to, and feel free to do the squash-and-merge operation too.

**Pull request merged**